Splits essentials.ext Permission

sasschary opened the pull request:
Takes essentials.ext and splits it into essentials.ext (Extinguish one's self) and essentials.ext.others (Extinguishing others). Adds a permission denied message if a user tries to extinguish another user if they do not have essentials.ext.others.

Fixes #2796

sasschary committed:
Splits essentials.ext Permission

Takes essentials.ext and splits it into essentials.ext (Extinguish one's self) and essentials.ext.others (Extinguishing others). Adds a permission denied message if a user tries to extinguish another user if they do not have essentials.ext.others

mdcfe commented:
This breaks existing permissions setups. I'd prefer either this is made configurable or `essentials.ext` grants the others permission by default to prevent breakage.

sasschary committed:
Makes essentials.ext.others configurable

Adds config option "ext-others-separate", which will decide whether or not essentials.ext should be a separate permission from essentials.ext.others.

sasschary commented:
Updated so that essentials.ext.others being a separate permission from essentials.ext is a configurable option.

caojohnny commented on Essentials/src/config.yml:
```diff
@@ -574,6 +574,11 @@ is-water-safe: false
 # You should only change this to false if you use Minecraft China.
 safe-usermap-names: true
 
+# Should essentials.ext be a separate oermission from essentials.ext.others?
+# If false, those with essentials.ext can also extinguish others with explicitly having essentials.ext.others.
```
- `oermission` -> `permission`
- `with` -> `without`?

mdcfe replied:
* `separate oermission from essentials.ext.others` -> `separate permission to essentials.ext.others`
* `If false, those with essentials.ext can also extinguish others with explicitly having essentials.ext.others.` -> `If false, players with essentials.ext can also extinguish others without having essentials.ext.others.`

mdcfe commented:
Looking back, I'd rather this was implemented without the config option. Server owners can set the permission themselves or we can make it a default child permission. Adding a config option is needlessly complex.

mdcfe commented:
Closing in favour of #2826, which does this in a way which is more consistent with existing EssentialsX commands.
